boards/arm/stm32l562e_dk/stm32l562e_dk.dts

```diff
@@ -31,8 +31,18 @@
 	cpu-power-states = <&stop0 &stop1 &stop2>;
 };
 
+&pinctrl {
+	usart1_lp_tx_pa9: usart1_lp_tx_pa9 {
+		pinmux = <STM32_PINMUX('A', 9, ANALOG)>;
+	};
+	usart1_lp_rx_pa10: usart1_lp_rx_pa10 {
+		pinmux = <STM32_PINMUX('A', 10, ANALOG)>;
+	};
+};
+
 &usart1 {
 	pinctrl-0 = <&usart1_tx_pa9 &usart1_rx_pa10>;
+	pinctrl-1 = <&usart1_lp_tx_pa9 &usart1_lp_rx_pa10>;
```

Comment:
I thik that we should introduce the `pinctrl-names` option as in Linux, i.e.
```devicetree
pinctrl-names = "default", "sleep";
```
C Macros used to obtain pinctrl information should refer to states and not indexes. We may need some devicetree scripts changes to handle that.

Reply:
Agreed. I don't think we need more device tree macros for that (at least from devicetree.h).
Everything should be there, what is missing is the `ST_STM32_DT_INST_PINCTRL_FOO` version.

Reply:
Let me know if you do decide you need something.

Reply:
Do we have the infrastructure to query by name on entries indexed like pinctrl, i.e. `pinctrl-0, pinctrl-1...pinctrl-N`? Can we get now the index `N` of an entry given its name, to then get the contents of `pinctrl-N`? I also think we need to support `pinctrl-N` and `pinctrl-names` in a generic way (we now repeat that on every binding).